mava/systems/jax/system.py

```diff
@@ -37,6 +37,17 @@ def __init__(self) -> None:
         # make config from build
         self._make_config()
 
+        # Enforce that design keys match component names
+        for key, value in self._design.get().items():
+            if key != value.name():
+                raise Exception(
+                    "Component '"
+                    + key
+                    + "' has mismatching name '"
+                    + value.name()
+                    + "'"
```

Comment:
Super pedantic, but fstring would make this a lot more readable: `f"Component '{key}' has mismatching name '{value.name}'"`

Reply:
Agreed, thanks @sash-a. Will change now